src/Kestrel.Core/Internal/HttpsConnectionAdapter.cs

```diff
@@ -158,6 +158,8 @@ public Task<IAdaptedConnection> OnConnectionAsync(ConnectionAdapterContext conte
                 if ((_options.HttpProtocols & HttpProtocols.Http2) != 0)
                 {
                     sslOptions.ApplicationProtocols.Add(SslApplicationProtocol.Http2);
+                    // https://tools.ietf.org/html/rfc7540#section-9.2.1
+                    sslOptions.AllowRenegotiation = false;
```

Comment:
I'm halfway tempted to disable this for all https connections. Do you know of any clients that actually renegotiate for any reason?

Reply:
I've mainly seen it used by the server to request a client certificate, but I wouldn't risk changing it for existing users right now.

>   Note that disabling
  renegotiation can result in long-lived connections becoming unusable
   due to limits on the number of messages the underlying cipher suite
   can encipher.